include/device.h

```diff
@@ -382,6 +382,13 @@ struct device {
 	 * device_required_handles_get().
 	 */
 	const device_handle_t *const handles;
+	/** optional pointer to handles supported by this device.
+	 *
+	 * This encodes a sequence of sets of device handles that
+	 * depends on this node.  The individual sets are extracted
+	 * with dedicated API, such as device_supported_handles_get().
+	 */
+	const device_handle_t *const supported_handles;
```

Comment:
is supported the right terminology here? is this supposed to mean "Dependent Handles"

Reply:
"supports" is the nomenclature used in devicetree.

Reply:
btw this field will go away. Just keeping it while https://github.com/zephyrproject-rtos/zephyr/pull/37836 is not merged.